### PR TITLE
Use acked task on creating and updating subscription

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Even though this doesn't help with the flakiness, we should use acked task in LR DDL operations, similar to other DDL operations.

### CODE REVIEW

File: server/src/main/java/io/crate/replication/logical/action/TransportCreateSubscriptionAction.java [hunk 1]
```diff
@@ -167,12 +168,14 @@ private static void checkVersionCompatibility(String tableFqn, Version subscribe
...
-        var future = new CompletableFuture<Void>();
+    private CompletableFuture<AcknowledgedResponse> submitClusterStateTask(CreateSubscriptionRequest request,
+                                                                           PublicationsStateAction.Response publicationsStateResponse) {
+
+        FutureActionListener<AcknowledgedResponse, AcknowledgedResponse> future = FutureActionListener.newInstance();
+
         clusterService.submitStateUpdateTask(
             source,
-            new ClusterStateUpdateTask() {
+            new AckedClusterStateUpdateTask<>(request, future) {
```
Comment (seut) [inline server/src/main/java/io/crate/replication/logical/action/TransportCreateSubscriptionAction.java:178]: Looks like we do not need any new intermediate future/listener, we can just return the `AckedClusterStateUpdateTask.completionFuture()` instead.

File: server/src/main/java/io/crate/replication/logical/action/UpdateSubscriptionAction.java [hunk 2]
```diff
@@ -169,7 +167,7 @@ protected ClusterBlockException checkBlock(Request request,
     }
 
 
-    public static class Request extends MasterNodeReadRequest<Request> {
+    public static class Request extends MasterNodeReadRequest<Request> implements AckedRequest {
```
Comment (seut) [inline server/src/main/java/io/crate/replication/logical/action/UpdateSubscriptionAction.java:170]: This looks initially wrong as this action is not only reading but writing. Correct fix imho is changing to extend from `AcknowledgedRequest`.
```suggestion
    public static class Request extends AcknowledgedRequest<Request> {
```

File: server/src/main/java/io/crate/replication/logical/action/UpdateSubscriptionAction.java [hunk 3]
```diff
@@ -191,5 +189,10 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeString(subscriptionName);
             subscription.writeTo(out);
         }
+
+        @Override
+        public TimeValue ackTimeout() {
+            return AcknowledgedRequest.DEFAULT_ACK_TIMEOUT;
+        }
```
Comment (seut) [inline server/src/main/java/io/crate/replication/logical/action/UpdateSubscriptionAction.java:196]: With the above change this is not needed I think.

File: server/src/test/java/io/crate/integrationtests/LogicalReplicationITest.java [hunk 4]
```diff
@@ -255,6 +255,9 @@ public void test_subscribing_to_unknown_publication_raises_error() throws Except
                                                      " CONNECTION '" + publisherConnectionUrl() + "' publication unknown_pub"))
             .isExactlyInstanceOf(PublicationUnknownException.class)
             .hasMessageContaining("Publication 'unknown_pub' unknown");
+
+        var response = executeOnSubscriber("SELECT * FROM pg_subscription");
+        assertThat(response).hasRowCount(0);
```
Comment (seut) [inline server/src/test/java/io/crate/integrationtests/LogicalReplicationITest.java:260]: Why this change, how is it related?

File: server/src/test/java/io/crate/integrationtests/LogicalReplicationITest.java [hunk 5]
```diff
@@ -299,6 +302,8 @@ public void test_subscribing_to_publication_while_table_exists_raises_error() th
             RelationAlreadyExists.class,
             () -> createSubscription("sub1", "pub1")
         );
+        var response = executeOnSubscriber("SELECT * FROM pg_subscription");
+        assertThat(response).hasRowCount(0);
```
Comment (seut) [inline server/src/test/java/io/crate/integrationtests/LogicalReplicationITest.java:306]: Same here, how is it related?

File: server/src/test/java/io/crate/integrationtests/LogicalReplicationITest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (seut) [reply server/src/test/java/io/crate/integrationtests/LogicalReplicationITest.java:260]: I do not agree as I think its not related  and any real statement execution costs time.

File: server/src/test/java/io/crate/replication/logical/action/TransportCreateSubscriptionActionTest.java [hunk 6]
```diff
@@ -83,12 +92,28 @@ private void subscribeToTablesWithVersion(FutureActionListener<AcknowledgedRespo
...
+            Version.CURRENT
+        );
+        ClusterState clusterState = ClusterState
+            .builder(ClusterName.DEFAULT)
+            .nodes(DiscoveryNodes.builder().add(dataNode).localNodeId(dataNode.getId()))
+            .build();
+        when(clusterService.state()).thenReturn(clusterState);
+
+        doAnswer(invocation -> responseFuture.complete(new AcknowledgedResponse(true)))
+            .when(clusterService).submitStateUpdateTask(anyString(), any());
```
Comment (seut) [inline server/src/test/java/io/crate/replication/logical/action/TransportCreateSubscriptionActionTest.java:115]: Why this change, how is it related?

File: server/src/test/java/io/crate/integrationtests/LogicalReplicationITest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/integrationtests/LogicalReplicationITest.java]: Removed new assertions

File: server/src/main/java/io/crate/replication/logical/action/TransportCreateSubscriptionAction.java (hunk 1 above)
Comment (BaurzhanSakhariev) [reply server/src/main/java/io/crate/replication/logical/action/TransportCreateSubscriptionAction.java]: Removed intermediate future in https://github.com/crate/crate/pull/14926/commits/d55d9e757c1594a0a61e349f00472bebe51767fe

File: server/src/test/java/io/crate/replication/logical/action/TransportCreateSubscriptionActionTest.java (hunk 6 above)
Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/replication/logical/action/TransportCreateSubscriptionActionTest.java:115]: Previously, non-mock `clusterService`, inherited from `CrateDummyClusterServiceUnitTest` used to work when submitting regular tasks. `clusterStateProcessed() ` is hit and future is completed.

`onAllNodesAcked` is never hit in unit tests - I think because we don't have any other nodes to ack? 

Because of that I made clusterService mock and then had to mock clusterService.state() as well.

**Review by seut — APPROVED** [review]
